src/pkg/bundle/bundle.go

```diff
@@ -81,7 +82,37 @@ func Create(b *Bundler, signature []byte) error {
 					}
 					// ensure media type is Zarf blob for layers in the bundle's root manifest
 					layerDesc.MediaType = oci.ZarfLayerMediaTypeBlob
+
+					// add package name annotations
+					annotations := make(map[string]string)
+					layerDesc.Annotations = annotations
+					layerDesc.Annotations[config.UDSPackageNameAnnotation] = pkg.Name
+
+					// If zarf package name has been obtained from zarf config, set the zarf package name annotation
+					if zarfPackageName != "" {
+						layerDesc.Annotations[config.ZarfPackageNameAnnotation] = zarfPackageName
+					}
+
 					rootManifest.Layers = append(rootManifest.Layers, layerDesc)
+					zarfLayerAdded = true
+				}
+				if layerDesc.MediaType == "application/vnd.zarf.config.v1+json" {
```

Comment:
This block can be refactored in a couple ways:
- Zarf has a constant for that zarf.config media type
- have a more specific variable name than `data`
- we can use Zarf's ConfigPartial on `data` to avoid the generic types